### PR TITLE
Minor improvements of the waypoint updater

### PR DESCRIPTION
**Pull request by idanov**
I've put back the optimisation in `publish()` since it makes the code almost twice as efficient.
I've also made the car stop before the stop line, thus making it harder to surpass the traffic light point in case of delays in the communication between the DBW and the simulator.

### CODE REVIEW

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 1]
```diff
@@ -73,7 +73,9 @@ def __init__(self):
 
     def publish(self):
         next_idx = self.find_next_waypoint()
-        if -1 < next_idx and not rospy.is_shutdown():
+        dist = self.dist_to_tl()
+        refresh = next_idx != self.next_idx or 10 > dist > -1
```
Comment (carlosgalvezp) [inline ros/src/waypoint_updater/waypoint_updater.py]: What if for some reason we stop passed the stop line? Then `dist` would either be negative or too large, thus not refreshing?

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 2]
```diff
@@ -237,7 +241,7 @@ def dist_to_tl(self):
     def decelerate(self, waypoints):
         dist = self.dist_to_tl()
         if self.tl_idx > - 1 and dist < LOOKAHEAD_WPS:
-            prev_idx = max(dist-2, 0)
+            prev_idx = max(dist-4, 0)
```
Comment (carlosgalvezp) [inline ros/src/waypoint_updater/waypoint_updater.py]: I think we should define this variable somewhere. It really has an important meaning: how many waypoints before the stop light we stop the car.

**Review by carlosgalvezp — COMMENTED** [review]
Nice! Just a small comment

**Review by carlosgalvezp — COMMENTED** [review]
I have just tested and the car didn't move after the light turned green... I really think we should not have these optimizations

**Comment by idanov** [comment]
@carlosgalvezp comments addressed, please merge it once approved.

**Review by carlosgalvezp — APPROVED** [review]
Awesome! Works really well, finally the car brakes on time! Thanks :)

**Pull request merged**